spec/models/mixins/supports_feature_mixin_spec.rb

```diff
@@ -150,18 +155,22 @@ def initialize(options = {})
     context "when the condition is met" do
       it "is supported on the class" do
         expect(SpecialPost.supports_fake?).to be true
+        expect(SpecialPost.supports_fake_with_block_syntax?).to be true
       end
 
       it "is supported on the instance" do
         expect(SpecialPost.new(:bribe => true).supports_fake?).to be true
+        expect(SpecialPost.new(:bribe => true).supports_fake_with_block_syntax?).to be true
       end
 
       it "gives no reason on the class" do
         expect(SpecialPost.unsupported_reason(:fake)).to be nil
+        expect(SpecialPost.unsupported_reason(:fake_with_block_syntax)).to be nil
       end
 
       it "gives no reason on the instance" do
         expect(SpecialPost.new(:bribe => true).unsupported_reason(:fake)).to be nil
+        expect(SpecialPost.new(:bribe => true).unsupported_reason(:fake_with_block_syntax)).to be nil
```

Comment:
answer: no, if the reason is nil (only really happens for the empty `supports :create` type methods), then a default reason is given.

Also, above, it would probably be better to use `if` over `||=`

```ruby
supports :shutdown do
  if !supports?(:control)
    unsupported_reason(:control) unless 
  elsif power_state == "on"
    _("The VM is not powered on")
  end
end
```